frontend/src/metabase/dashboard/components/DashboardHeader/DashboardHeaderButtonRow.tsx

```diff
@@ -0,0 +1,315 @@
+    enabled: ({ isEditing }) => isEditing,
+  },
+  [BUTTON_CONFIG.ADD_LINK_CARD]: {
+    component: () => <AddLinkCardButton />,
+    enabled: ({ isEditing }) => isEditing,
+  },
+  [BUTTON_CONFIG.ADD_SECTION]: {
+    component: () => <AddSectionButton />,
+    enabled: ({ isEditing }) => isEditing,
+  },
+  [BUTTON_CONFIG.ADD_TEMPORAL_UNIT]: {
+    component: () => <AddTemporalUnitButton />,
+    enabled: ({ isEditing }) => isEditing,
+  },
+  [BUTTON_CONFIG.ADD_FILTER_PARAMETER]: {
+    component: () => <AddFilterParameterButton />,
+    enabled: ({ isEditing }) => isEditing,
+  },
+  [BUTTON_CONFIG.ADD_ACTION_ELEMENT]: {
+    component: () => <AddActionElementButton />,
+    enabled: ({ isEditing, canEdit, hasModelActionsEnabled }) =>
+      isEditing && canEdit && hasModelActionsEnabled,
+  },
+  [BUTTON_CONFIG.EXTRA_EDIT_BUTTONS_MENU]: {
+    component: () => <ExtraEditButtonsMenu />,
+    enabled: ({ isEditing }) => isEditing,
+  },
+  [BUTTON_CONFIG.COPY_ANALYTICS_DASHBOARD]: {
+    component: () => <CopyAnalyticsDashboardButton />,
+    enabled: ({ isAnalyticsDashboard }) => isAnalyticsDashboard,
+  },
+  [BUTTON_CONFIG.EDIT_DASHBOARD]: {
+    component: ({ onRefreshPeriodChange }) => (
+      <EditDashboardButton onRefreshPeriodChange={onRefreshPeriodChange} />
+    ),
+    enabled: ({ isFullscreen, isEditing, canEdit }) =>
+      !isFullscreen && !isEditing && canEdit,
+  },
+  [BUTTON_CONFIG.DASHBOARD_SUBSCRIPTION]: {
+    component: () => <DashboardSubscriptionButton />,
+    enabled: ({
+      dashboard,
+      canManageSubscriptions,
+      formInput,
+      isAdmin,
+      isEditing,
+      isFullscreen,
+    }) =>
+      shouldRenderSubscriptionButton({
+        dashboard,
+        canManageSubscriptions,
+        formInput,
+        isAdmin,
+        isEditing,
+        isFullscreen,
+      }),
+  },
+  [BUTTON_CONFIG.DASHBOARD_EMBED_ACTION]: {
+    component: () => <DashboardEmbedAction />,
+    enabled: ({ dashboard, isPublic, isEmpty }) =>
+      !isEmpty && !isPublic && dashboard && !dashboard?.archived,
+  },
+  [BUTTON_CONFIG.REFRESH_WIDGET]: {
+    component: ({
+      refreshPeriod,
+      setRefreshElapsedHook,
+      onRefreshPeriodChange,
+    }) => (
+      <RefreshWidgetButton
+        period={refreshPeriod}
+        setRefreshElapsedHook={setRefreshElapsedHook}
+        onChangePeriod={onRefreshPeriodChange}
+      />
+    ),
+    enabled: ({ dashboard, isEmpty }) => !isEmpty && !dashboard?.archived,
+  },
+  [BUTTON_CONFIG.NIGHT_MODE_TOGGLE]: {
+    component: ({ isNightMode, onNightModeChange }) => (
+      <NightModeToggleButton
+        isNightMode={isNightMode}
+        onNightModeChange={onNightModeChange}
+      />
+    ),
+    enabled: ({
+      isFullscreen,
+      dashboard,
+      hasNightModeToggle,
+      onNightModeChange,
+    }) =>
+      Boolean(
+        isFullscreen &&
+          !dashboard.archived &&
+          hasNightModeToggle &&
+          !!onNightModeChange,
```

Comment:
you likely don't need to cast this since it should be coerced by `&&` operator already.

```suggestion
          onNightModeChange,
```